### PR TITLE
Apply external ignored_labels to C-IL for detection

### PR DESCRIPTION
**Pull request by harimkang**
### This PR includes :
- Incoming ignored_labels from OTE is received from 'img_metas' to create an ignored mask
- Deliver ignored masks to loss to use as a mask (Modify the forward & loss function on the head side to receive the ignored_mask)

### CODE REVIEW

File: mpa/modules/models/heads/cross_dataset_detector_head.py [hunk 1]
```diff
@@ -236,4 +217,15 @@ def vfnet_to_atss_targets(self,
...
-        return labels_list, label_weights, bbox_targets_list, bbox_weights
+        return labels_list, label_weights, bbox_targets_list, bbox_weights, ignored_masks
+
+    def get_ignored_masks(self, img_metas, all_labels):
+        ignored_masks = []
+        for i, meta in enumerate(img_metas):
+            mask = torch.Tensor([1 for _ in range(self.num_classes)])
+            if 'ignored_labels' in meta and meta['ignored_labels']:
+                mask[meta['ignored_labels']] = 0
+            mask = mask.repeat(len(all_labels[i]), 1)
```
Comment (chuneuny-emily) [inline mpa/modules/models/heads/cross_dataset_detector_head.py:228]: Do you need mask.repeat code? What is the purpose?

Comment (harimkang) [reply mpa/modules/models/heads/cross_dataset_detector_head.py:228]: @chuneuny-emily Yes, this is a function that takes ignored_labels and makes them a mask that will be used for loss. Because information about which images each label belongs to is not passed to loss, the ignored_labels are replaced with masks in the head.

Comment (chuneuny-emily) [reply mpa/modules/models/heads/cross_dataset_detector_head.py:228]: > @chuneuny-emily Yes, this is a function that takes ignored_labels and makes them a mask that will be used for loss. Because information about which images each label belongs to is not passed to loss, the ignored_labels are replaced with masks in the head.

I understood.

**Review by chuneuny-emily — APPROVED** [review]
It looks good for me.

File: mpa/modules/models/heads/custom_vfnet_head.py [hunk 2]
```diff
@@ -154,19 +160,22 @@ def loss(self,
...
             if isinstance(self.loss_cls, CrossSigmoidFocalLoss):
                 loss_cls = self.loss_cls(
                     flatten_cls_scores,
                     cls_iou_targets,
-                    weight=label_weights.unsqueeze(-1),
+                    weight=label_weights,
                     avg_factor=num_pos_avg_per_gpu,
                     use_vfl=self.use_vfl,
-                    use_weight=self.use_atss)
+                    use_weight=self.use_atss,
```
Comment (goodsong81) [inline mpa/modules/models/heads/custom_vfnet_head.py]: Could you explain why?

**Review by goodsong81 — APPROVED** [review]
LGTM. Could you answer the question in the comment?

File: mpa/modules/models/heads/custom_vfnet_head.py (hunk 2 above)
Comment (harimkang) [reply mpa/modules/models/heads/custom_vfnet_head.py]: I missed that part and didn't push it. I removed it.

File: mpa/modules/models/heads/cross_dataset_detector_head.py [hunk 3]
```diff
@@ -236,4 +217,15 @@ def vfnet_to_atss_targets(self,
         ]
         label_weights = torch.cat(label_weights_list)
         bbox_weights = torch.cat(bbox_weights_list)
-        return labels_list, label_weights, bbox_targets_list, bbox_weights
+        return labels_list, label_weights, bbox_targets_list, bbox_weights, ignored_masks
+
+    def get_ignored_masks(self, img_metas, all_labels):
```
Comment (goodsong81) [inline mpa/modules/models/heads/cross_dataset_detector_head.py]: Is this 'ignored' mask or 'valid' mask?
Actually, valid_label_mask would be better in terms of clarification.

**Review by goodsong81 — CHANGES_REQUESTED** [review]

File: mpa/modules/models/heads/cross_dataset_detector_head.py (hunk 3 above)
Comment (harimkang) [reply mpa/modules/models/heads/cross_dataset_detector_head.py]: The ignored_masks were modified to valid_label_mask.

**Review by goodsong81 — APPROVED** [review]
